test/integration/integration_test.go

```diff
@@ -200,3 +173,68 @@ func testMultipleIngresses(t *testing.T) {
 		}
 	}
 }
+
+type TestConfig struct {
+	operatorNamespace string
+	clusterName       string
+
+	kubeConfig string
+
+	t *testing.T
+}
+
+func NewTestConfig(t *testing.T) *TestConfig {
+	config := &TestConfig{t: t}
+
+	// Check prerequisites
+	kubeConfig := os.Getenv("KUBECONFIG")
+	if len(kubeConfig) == 0 {
+		t.Fatalf("KUBECONFIG is required")
+	}
+	// The operator-sdk uses KUBERNETES_CONFIG...
+	os.Setenv("KUBERNETES_CONFIG", kubeConfig)
+	config.kubeConfig = kubeConfig
+
+	clusterName := os.Getenv("CLUSTER_NAME")
+	if len(clusterName) == 0 {
+		t.Fatalf("CLUSTER_NAME is required")
+	}
+	config.clusterName = clusterName
+
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		namespace = "default"
```

Comment:
Just a note: I want this to go away ASAP; we should be applying /manifests which will put the operator into its own namespace with all the requisite RBAC resources rather than piggybacking on `default`.

Reply:
Sure. Till we land that change, this will not force users to specify another env (WATCH_NAMESPACE)